thunder/core/proxies.py

```diff
@@ -1251,7 +1256,7 @@ def _infer_tensor_properties(
     else:
         # deferred computation of numel
         # TODO: similar to how `shape` is handled, this should be CSE or lifted for efficiency
-        _numel = lambda tp: reduce(operator.mul, tp.shape, 1)
+        _numel = lambda *args: reduce(operator.mul, _shape, 1)
```

Comment:
AFAICT `_shape` can be `None` here. Do we need to guard against that?

Reply:
Good call. Let me throw a meaningful error message for that.

Reply:
Seems to be there now.